org.jacoco.core.test/src/org/jacoco/core/internal/analysis/MethodAnalyzerTest.java

```diff
@@ -43,7 +47,30 @@
 	@Before
 	public void setup() {
 		nextProbeId = 0;
-		method = new MethodNode();
+		method = new MethodNode(InstrSupport.ASM_API_VERSION) {
```

Comment:
@marchof Nice catch! Actually I completely forgot about existance of `CheckMethodAdapter`, so will reconsider/rework this - feel free to bloc merging by marking review as "requires changes" :wink: